cabal.project

```diff
@@ -103,6 +103,10 @@ package shelley-spec-ledger
 package cardano-ledger
   tests: False
 
+-- The two following one-liners will restore / cut off the remainder of this file (for nix-shell users):
+-- git checkout HEAD "$(git rev-parse --show-toplevel)"/cabal.project
+-- sed -ni '1,/--- 8< ---/ p' "$(git rev-parse --show-toplevel)"/cabal.project
+-- Please do not put any `source-repository-package` clause above this line.
```

Comment:
This solution seems fragile and prone to error and people mistakenly submitting PRs with un-intended changes.

Please find a better way.

Reply:
@erikd, two things:

1. There is a better way, and Oleg Grenrus is working on that -- this will allow transparent reuse of `haskell.nix` packages by Cabal's externals mechanism.

2. When you say it's fragile and error prone -- do you consider how much of an improvement of an entirely manual process this is?

3. Even if we accept that it is fragile and error-prone, it only affects users of the said workflow.   Surely we can afford Nix people deal with their problems?

Reply:
We have enough horrible hacky stuff in the repository already so what is one more horrible hack?

Reply:
And by the way, should I mention that for the users of the said workflow it's a constant pain?